cpp/support/utils.h

```diff
@@ -109,21 +130,22 @@ template <typename T, typename E = std::runtime_error>
 class Result {
  private:
   static_assert(!std::is_same_v<T, E>, "T and E cannot be the same type");
+  using OkTag = std::in_place_type_t<T>;
+  using ErrTag = std::in_place_type_t<E>;
```

Comment:
```suggestion
  using OkTag = std::in_place_type_t<T>;
  using ErrTag = std::in_place_type_t<E>;
```

We don't need this. Just using inplace type is ok